src/metric/event.rs

```diff
@@ -22,6 +29,8 @@ pub enum Event {
     /// more events will appear.  It is expected that after receiving this
     /// marker the given source will exit cleanly.
     Shutdown,
+    /// Raw, encoded bytes.
+    Raw(Encoding, Vec<u8>),
```

Comment:
I will 100% forget the ordering here. How do you feel about using named fields, like: `Raw { encoding: Encoding, bytes: Vec<u8> }`? Then we can refer to them by name, match by name.

Reply:
Fixed.